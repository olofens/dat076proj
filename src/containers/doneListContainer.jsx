import React from "react";
import ReactDOM from "react-dom";
import { connect } from 'react-redux';
import { bindActionCreators } from "redux";
import RightTask from "../components/rightTask.jsx";
import { doneClick } from "../actions/index.js"
import DoneListComponent from "../components/doneListComponent.jsx"

class DoneListContainer extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <DoneListComponent
        tasks={this.props.tasks}
        click={this.props.doneClick}
      />
    );
  }
}

function mapStateToProps(state) {
  return {
    tasks: state.doneTasks
  }
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators({ doneClick: doneClick }, dispatch)
}

export default connect(mapStateToProps, matchDispatchToProps)(DoneListContainer);
