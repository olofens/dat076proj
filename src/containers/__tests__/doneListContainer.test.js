import React from 'react';
import { shallow } from 'enzyme';
import doneListContainer from '../doneListContainer.jsx';


describe('doneListContainer', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<doneListContainer debug />);
  
    expect(component).toMatchSnapshot();
  });
});