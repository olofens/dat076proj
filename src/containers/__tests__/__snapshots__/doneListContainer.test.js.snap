// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doneListContainer should render correctly in "debug" mode 1`] = `
<doneListContainer
  debug={true}
/>
`;
